jung-visualization/src/main/java/edu/uci/ics/jung/visualization/DefaultVisualizationModel.java

```diff
@@ -65,37 +66,43 @@ public void stateChanged(ChangeEvent e) {
             }
           };
     }
-    setGraphLayout(layout, d);
+    setLayoutMediator(new LayoutMediator(network, layout), d);
   }
 
   /**
    * Removes the current graph layout, and adds a new one.
    *
-   * @param layout the new layout to use
+   * @param layoutMediator the new layoutMediator to use
    * @param viewSize the size of the View that will display this layout
    */
-  public void setGraphLayout(Layout<N> layout, Dimension viewSize) {
+  public void setLayoutMediator(LayoutMediator<N, E> layoutMediator, Dimension viewSize) {
+
+    Layout<N> currentLayout = this.layoutMediator != null ? this.layoutMediator.getLayout() : null;
     // remove listener from old layout
-    if (this.layout != null && this.layout instanceof ChangeEventSupport) {
-      ((ChangeEventSupport) this.layout).removeChangeListener(changeListener);
+    if (currentLayout != null && currentLayout instanceof ChangeEventSupport) {
+      ((ChangeEventSupport) currentLayout).removeChangeListener(changeListener);
     }
+    Layout<N> newLayout = layoutMediator.getLayout();
+
     // set to new layout
-    if (layout instanceof ChangeEventSupport) {
-      this.layout = layout;
+    if (newLayout instanceof ChangeEventSupport) {
+      this.layoutMediator = layoutMediator;
     } else {
-      this.layout = new ObservableCachingLayout<N>(network.asGraph(), layout);
+      newLayout = new ObservableCachingLayout<N, E>(layoutMediator.getNetwork(), newLayout);
+      this.layoutMediator = new LayoutMediator(layoutMediator.getNetwork(), newLayout);
     }
 
-    ((ChangeEventSupport) this.layout).addChangeListener(changeListener);
+    ((ChangeEventSupport) newLayout).addChangeListener(changeListener);
 
     if (viewSize == null) {
       viewSize = new Dimension(600, 600);
     }
-    Dimension layoutSize = layout.getSize();
+    Dimension layoutSize = newLayout.getSize();
     // if the layout has NOT been initialized yet, initialize its size
     // now to the size of the VisualizationViewer window
     if (layoutSize == null) {
-      layout.setSize(viewSize);
+      newLayout.setSize(viewSize);
+      //      newLayout.setSize(viewSize);
```

Comment:
I believe this comment has very little use. Would you please consider removing it.